### PR TITLE
[MO] - [system test] -> oauth authentification with instrospection endpoint

### PR DESCRIPTION
**Pull request by see-quick**
### Type of change

- Bugfix
- Enhancement / new feature

### Description

This PR resolving bug with setting oauth authentification with help of introspection endpoint. When i deploy Kafka with configuration on introspection i do not want to set also two properties which are not relevant:
```
1) JwksRefreshSeconds
2) JwksExpirySeconds
```

This commit is reference from oauth authorization, so after authorization PR will be merged i will do a rebase and this lines should disappear. 

### Checklist

- [x] Write tests
- [x] Make sure all tests pass

### CODE REVIEW

**Comment by Seequick1** [comment]
@strimzi-ci run tests profile=regression testcase=io.strimzi.systemtest.oauth.*ST test_only

**Comment by Seequick1** [comment]
@strimzi-ci run tests profile=regression testcase=io.strimzi.systemtest.oauth.*ST

File: systemtest/src/test/java/io/strimzi/systemtest/oauth/OauthBaseST.java [hunk 1]
```diff
@@ -48,11 +48,14 @@
 
     protected static final String OAUTH_CLIENT_NAME = "hello-world-producer";
     protected static final String OAUTH_CLIENT_SECRET = "hello-world-producer-secret";
+    protected static final String OAUTH_KAFKA_CLIENT_NAME = "kafka-broker";
+
```
Comment (tombentley) [inline systemtest/src/test/java/io/strimzi/systemtest/oauth/OauthBaseST.java]: I don't think we need two blank lines.

**Comment by Seequick1** [comment]
@strimzi-ci run tests profile=regression testcase=OauthPlainST#testIntrospectionEndpointWithPlainCommunication

**Comment by see-quick** [comment]
@strimzi-ci run tests profile=regression testcase=OauthPlainST#testIntrospectionEndpointWithPlainCommunication

**Comment by see-quick** [comment]
because when we i use deploy broker without using refresh and expire jwks it will trigger an error, where will say that these properties needs to be more than 1. So another should be to set it to default value as you proposed previously like 
```
return jwksRefreshSeconds == null ? Integer.valueOf(300) : jwksRefreshSeconds;
```
same with expire jwks.

**Comment by tombentley** [comment]
> it will trigger an error,

Is that an error from the apiserver, or something else? Because having a minimum should not imply that a property is required. But if the apiserver imposes (and it sounds like it is) that then there's nothing we can do about it.

File: api/src/main/java/io/strimzi/api/kafka/model/listener/KafkaListenerAuthenticationOAuth.java [hunk 2]
```diff
@@ -95,8 +93,6 @@ public void setJwksEndpointUri(String jwksEndpointUri) {
     @Description("Configures how often are the JWKS certificates refreshed. " +
             "The refresh interval has to be at least 60 seconds shorter then the expiry interval specified in `jwksExpirySeconds`. " +
             "Defaults to 300 seconds.")
-    @Minimum(1)
```
Comment (scholzj) [inline api/src/main/java/io/strimzi/api/kafka/model/listener/KafkaListenerAuthenticationOAuth.java]: I do not really understand the problem here. With regular YAMLs, when the field is not present, the minimum value is ignored and doesn't cause any issues.

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaCluster.java [hunk 3]
```diff
@@ -1748,19 +1751,27 @@ private static void validateOauth(KafkaListenerAuthenticationOAuth oAuth, String
             throw new InvalidResourceException(listener + ": Valid Issuer URI has to be specified");
         }
 
+        if (oAuth.getJwksEndpointUri() != null && (oAuth.getJwksExpirySeconds() != null || oAuth.getJwksRefreshSeconds() != null)) {
```
Comment (scholzj) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaCluster.java]: I do not really get it why did you added this condition. I do not think it is needed. You can use JWKS endpoint without specifying expire seconds or refeshs seconds. They have defaults.

**Pull request merged**